File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/data/api/interceptors/AuthInterceptor.java
package com.crpr.androidcinema.data.api.interceptors;

import java.io.IOException;

import okhttp3.HttpUrl;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * Created by claudioribeiro on 09/07/16.
 */
public class AuthInterceptor implements Interceptor {

    private final String api_key;

    public AuthInterceptor(String apiKey){
        this.api_key = apiKey;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request request = chain.request();

        HttpUrl url = request.url()
                        .newBuilder()
                        .addQueryParameter("api_key",this.api_key)
                        .build();

        request = request.newBuilder().url(url).build();

        return chain.proceed(request);
    }
}

File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/data/api/interceptors/AuthInterceptor.kt
package com.crpr.androidcinema.data.api.interceptors

import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

/**
 * Created by claudioribeiro on 09/07/16.
 */
class AuthInterceptor(private val api_key: String) : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        var request = chain.request()

        val url = request.url()
                .newBuilder()
                .addQueryParameter("api_key", this.api_key)
                .build()

        request = request.newBuilder().url(url).build()

        return chain.proceed(request)
    }
}
